astroid/nodes/node_classes.py

```diff
@@ -5006,7 +5007,9 @@ def postinit(self, target: NodeNG, value: NodeNG) -> None:
     See astroid/protocols.py for actual implementation.
     """
 
-    def frame(self):
+    def frame(
+        self, *, future: Literal[None, True] = None
+    ) -> Union["nodes.FunctionDef", "nodes.Module", "nodes.ClassDef", "nodes.Lambda"]:
```

Comment:
Seems like we want ahead and raised `ParentMissingError` here without any deprecation.. Oh well, damage is done already 😄

Reply:
Yeah.